web/src/components/TtsContainer.tsx

```diff
@@ -0,0 +1,172 @@
+import EasySpeech from 'easy-speech'
+import React, { ReactElement, useEffect, useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import { MAX_TITLE_DISPLAY_CHARS } from 'shared'
+import { truncate } from 'shared/utils/getExcerpt'
+
+import useTtsPlayer from '../hooks/useTtsPlayer'
+import { reportError } from '../utils/sentry'
+import TtsPlayer from './TtsPlayer'
+
+type TtsContainerProps = {
+  languageCode: string
+}
+
+const TtsContainer = ({ languageCode }: TtsContainerProps): ReactElement | null => {
+  const { t } = useTranslation('layout')
+  const [isPlaying, setIsPlaying] = useState<boolean>(false)
+  const { sentences, setSentences, visible, setVisible, enabled } = useTtsPlayer()
+  const [currentSentencesIndex, setCurrentSentenceIndex] = useState<number>(0)
+  const [showHelpModal, setShowHelpModal] = useState(false)
+  const title = sentences[0] || t('nothingToRead')
+  const shortTitle = truncate(title, { maxChars: MAX_TITLE_DISPLAY_CHARS })
+  const userAgent = navigator.userAgent.toLowerCase()
+  const isAndroid = /android/i.test(userAgent)
+  const isFirefoxAndLinux = userAgent.includes('firefox') && userAgent.includes('linux')
+  const onEndGuard = useRef(true)
+  const fallbackTimer = 1000
+
+  useEffect(() => {
+    if (!enabled && !visible) {
+      return
+    }
+
+    EasySpeech.init({ maxTimeout: 5000, interval: 250 }).catch(reportError)
+  }, [enabled, visible])
+
+  const resetOnEnd = () => {
+    setCurrentSentenceIndex(0)
+    setIsPlaying(false)
+    onEndGuard.current = false
+  }
+
+  const stop = () => {
+    try {
+      EasySpeech.cancel()
+      onEndGuard.current = false
+    } catch (e) {
+      reportError(e)
+    }
+  }
+
+  const play = async (index = currentSentencesIndex) => {
+    try {
+      const voices = EasySpeech.voices()
+      const selectedVoice = voices.find((voice: SpeechSynthesisVoice) => voice.lang.startsWith(languageCode))
+
+      if (selectedVoice == null || EasySpeech.status().status === 'created') {
+        setSentences([])
+        setShowHelpModal(true)
+        return
+      }
+
+      await EasySpeech.speak({
+        text: String(sentences[index]),
+        voice: selectedVoice,
+        volume: 0.6,
```

Comment:
we should add a rate here. For me on firefox its super fast and on chrome its okay with default rate.
It doesn't look like the default is for both browsers the same
`rate: 0.8`
worked good for me